agent/checks/os_service_windows.go

```diff
@@ -0,0 +1,114 @@
+//go:build windows
+// +build windows
+
+package checks
+
+import (
+	"fmt"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+var (
+	win windowsSystem = windowsOS{}
+)
+
+type OSServiceClient struct {
+	scHandle windows.Handle
+}
+
+func NewOSServiceClient() (*OSServiceClient, error) {
+	var s *uint16
+	scHandle, err := win.OpenSCManager(s, nil, windows.SC_MANAGER_CONNECT)
+
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to service manager: %w", err)
+	}
+
+	return &OSServiceClient{
+		scHandle: scHandle,
+	}, nil
+}
+
+func (client *OSServiceClient) Check(serviceName string) (err error) {
+	var isHealthy bool
+
+	m := win.getWindowsSvcMgr(client.scHandle)
+	defer func() {
+		errDisconnect := m.Disconnect()
+		if isHealthy || errDisconnect == nil || err != nil {
+			return
+		}
+		//unreachable at the moment but we might want to log this error. leaving here for code-review
+		err = errDisconnect
+	}()
```

Comment:
From a quick glance I think the fail conditions for Disconnect and Close are unlikely and can be ignored. 

In the worst case scenario the process holding the ServiceManager open doesn't get cleaned up but if the OS is degraded to the point where it can't close there may be worse problems 😅 

I'd vote for a simple defer but it's not a strong opinion by any means.

Reply:
Same comment here as the one about `getWindowsSvc`. What would happen if `OpenService`  fail and we execute Disconnect anyway?